tests/test_make.py

```diff
@@ -527,15 +525,12 @@ class A:
         desert.schema_class(A)
 
 
-@pytest.mark.skipif(
-    sys.version_info[:2] <= (3, 6), reason="3.6 has isinstance(t.Sequence[int], type)."
-)
```

Comment:
This seems like it is about dropping 3.6 support?  Maybe?  That would be a separate PR.  Or, maybe I'm misreading this.

Reply:
Nope, that was an accident. I'll fix it.

Reply:
We're adding support for `Sequence` so I figured we needed to change it to something else. I didn't realize that would break 3.6.

Reply:
Alrighty, changing the bit inside the test makes sense but i think the skip ends up still applying?

Reply:
Maybe I'm misunderstanding this -- what exactly is this skip avoiding?

Reply:
I don't really remember so let me write out my exploration here.  To start with, in 3.6 we get:

https://github.com/python-desert/desert/runs/4645282988?check_suite_focus=true#step:5:147
```python-traceback
______________________ test_raise_unknown_generic[attrs] _______________________
tests/test_make.py:543: in test_raise_unknown_generic
    desert.schema_class(A)
.tox/py36/lib/python3.6/site-packages/desert/__init__.py:47: in schema_class
    return desert._make.class_schema(cls, meta=meta)
.tox/py36/lib/python3.6/site-packages/desert/_make.py:130: in class_schema
    field.metadata,
.tox/py36/lib/python3.6/site-packages/desert/_make.py:313: in field_for_schema
    nested = forward_reference or class_schema(typ)
.tox/py36/lib/python3.6/site-packages/desert/_make.py:120: in class_schema
    raise desert.exceptions.NotAnAttrsClassOrDataclass(clazz)
E   desert.exceptions.NotAnAttrsClassOrDataclass: tests.test_make.UnknownGeneric[int]
```
```python-traceback
___________________ test_raise_unknown_generic[dataclasses] ____________________
tests/test_make.py:543: in test_raise_unknown_generic
    desert.schema_class(A)
.tox/py36/lib/python3.6/site-packages/desert/__init__.py:47: in schema_class
    return desert._make.class_schema(cls, meta=meta)
.tox/py36/lib/python3.6/site-packages/desert/_make.py:130: in class_schema
    field.metadata,
.tox/py36/lib/python3.6/site-packages/desert/_make.py:313: in field_for_schema
    nested = forward_reference or class_schema(typ)
.tox/py36/lib/python3.6/site-packages/desert/_make.py:120: in class_schema
    raise desert.exceptions.NotAnAttrsClassOrDataclass(clazz)
E   desert.exceptions.NotAnAttrsClassOrDataclass: tests.test_make.UnknownGeneric[int]
```

It was originally added in https://github.com/python-desert/desert/commit/d61f8db212112d5c40a587252dd558826e558fb1 and relates to the following code.

https://github.com/python-desert/desert/blob/32d4f6863886b8628ad0dcd670d51f69b6dd5c3b/src/desert/_make.py#L104-L117

Python 3.6 does treat the `isinstance()` check differently.

```console
$ for v in 3.6 3.7 3.8 3.9 3.10; do python$v -c "import typing; print('$v', isinstance(typing.Sequence, type))"; done
3.6 True
3.7 False
3.8 False
3.9 False
3.10 False
```

I haven't thought through the implications or how we might work around this to fix 3.6, but given that 3.6 is EOL now I think just adding the skip back until we remove 3.6 is probably the easiest path forward.

Once you get one submission merged GitHub will run builds for you without approval so you can see this stuff without waiting for our approval.  I guess awhile back they started running into too many PRs that were doing cryptocurrency mining or something so they locked down first-time contributors from having their builds run automatically.

Reply:
> I haven't thought through the implications or how we might work around this to fix 3.6, but given that 3.6 is EOL now I think just adding the skip back until we remove 3.6 is probably the easiest path forward.

Sounds good.